src/Raven.Studio/typescript/viewmodels/common/notificationCenter/detailViewer/operations/smugglerDatabaseDetails.ts

```diff
@@ -300,19 +305,42 @@ class smugglerDatabaseDetails extends abstractOperationDetails {
             }
         }
 
+        let processingSpeedText = smugglerDatabaseDetails.ProcessingText;
         const isDocuments = name === "Documents";
-        let processingSpeedText = "Processing";
+
         const skippedCount = isDocuments ? (item as Raven.Client.Documents.Smuggler.SmugglerProgressBase.CountsWithSkippedCountAndLastEtag).SkippedCount : 0;
-        if (isDocuments) {
-            const docsCount = item.ReadCount + skippedCount + item.ErroredCount;
-            if (docsCount === this.lastDocsCount) {
-                processingSpeedText = this.lastProcessingSpeedText;
-            } else {
-                this.lastDocsCount = docsCount;
-                const processingSpeed = this.calculateProcessingSpeed(item.ReadCount + skippedCount + item.ErroredCount);
-                if (processingSpeed > 0) {
-                    processingSpeedText = this.lastProcessingSpeedText = processingSpeed.toLocaleString() + " docs / sec";
-                }
+        
+        if (item.StartTime) {
+            const itemsCount = item.ReadCount + skippedCount + item.ErroredCount;
+            
+            switch (name) {
+                case "Documents":
+                    if (itemsCount === this.lastDocsCount) {
```

Comment:
same here - put that logic to smugger item and call single method  instead of repeating code (the same code) for each item.

Reply:
ok - refactored